frontend/src/features/projects/dispose/ProjectDisposeLayout.tsx

```diff
@@ -56,15 +46,11 @@ const ProjectDisposeLayout = ({ match, location }: { match: Match; location: Loc
   ) => {
     //if we are at the most recent incomplete step, update status and project.
     if (project?.statusId === currentStatus.id) {
-      if (
-        nextStepCode === ReviewWorkflowStatus.PropertyReview ||
-        nextStepCode === ReviewWorkflowStatus.ExemptionReview
-      ) {
-        history.push('/project/completed');
-      }
       return dispatch(updateWorkflowStatus(project, nextStepCode, workflowStatusCode) as any)
         .then((project: IProject) => {
-          goToNextStep(project);
+          if (goToNextStep(project) === undefined) {
+            history.push('/project/completed');
```

Comment:
this transition now only occurs after a successful update, previously it could occur even if the update wasn't successful.